qemu/tests/rh_kselftests_vm.py

```diff
@@ -41,7 +41,9 @@ def run(test, params, env):
         s, o = session.cmd_status_output(tests_execution_cmd, 180)
         test.log.info("The selftests results: %s", o)
 
-        summary = re.findall(r"\# SUMMARY.+", o)
+        summary_pattern = params.get("summary_pattern")
+        pattern = rf"{summary_pattern}"
+        summary = re.findall(pattern, o)
         num_failed_tests = int(re.findall(r"FAIL\=\d+", str(summary))[0].split("=")[1])
```

Comment:
what about using a single line pattern to get the count of failed cases, e.g. 
>>> o = re.search(r"SUMMARY:.+FAIL=(?P<fail>\d+?)", s)
>>> o.group("fail")
'0'

Just an example, please re-consider the pattern string
Then we dont need two findall calls

Reply:
Hello @zhencliu could you provide how does it work this implementation? I've quickly tried the following:
```python
import re

s = """
SUMMARY: PASS=4 SKIP=0 FAIL=0
"""

o = re.search(r"SUMMARY:.+FAIL=(?P\d+?)", s)
o.group("fail")
```
And I get: `re.error: unknown extension ?P\d at position 16`

Reply:
HAHA, github 'ate' <fail>
re.search(r"SUMMARY:.+FAIL=(?P<fail>\d+?)", s)

Reply:
@zhencliu got it code updated !
Since this approach can be used for skipped tests, updated that part as well